lib/components/src/TabsBar/TabsButton.tsx

```diff
@@ -0,0 +1,136 @@
+import { styled } from '@storybook/theming';
+import React, { forwardRef, HTMLAttributes, ReactNode, RefObject } from 'react';
+import { Icons, IconsProps } from '../icon/icon';
+
+export type TabsButtonProps = {
+  adornmentEnd?: ReactNode;
+  adornmentStart?: ReactNode;
+  active?: boolean;
+  activeColor?: string;
+  textColor?: string;
+  icon?: IconsProps['icon'];
+  Icon?: ReactNode;
+  narrow?: boolean;
+  label?: ReactNode;
+  LabelProps?: HTMLAttributes<HTMLSpanElement>;
+} & HTMLAttributes<HTMLButtonElement>;
+
+export const TabsButton = forwardRef(
+  (props: TabsButtonProps, ref: RefObject<HTMLButtonElement>) => {
+    const {
+      active,
+      adornmentStart,
+      adornmentEnd,
+      textColor,
+      icon,
+      Icon,
+      label,
+      narrow,
+      activeColor,
+      LabelProps = {},
+      ...rest
+    } = props;
+    let ButtonIcon = Icon;
+
+    if (icon) {
+      ButtonIcon = <Icons icon={icon} />;
+    }
+
+    return (
+      <Wrapper
+        data-sb-tabs-button=""
+        active={active}
+        narrow={narrow}
+        textColor={textColor}
+        activeColor={activeColor}
+        ref={ref}
+        {...rest}
+      >
+        {ButtonIcon && (
+          <IconWrapper data-sb-tabsbutton-icon="" hasLabel={label !== undefined}>
+            {ButtonIcon}
+          </IconWrapper>
+        )}
+        {adornmentStart}
+        {label && (
+          <span data-sb-tabsbutton-label="" {...LabelProps}>
+            {label}
+          </span>
+        )}
+        {adornmentEnd}
+      </Wrapper>
+    );
+  }
+);
+
+interface WrapperProps {
+  active: boolean;
+  textColor: string;
+  activeColor: string;
+  narrow: boolean;
+}
+
+const Wrapper = styled.button<WrapperProps>(
```

Comment:
I don't think `Wrapper` is the best name of this, this is actually the Button, and what's called `TabsButton` is the Wrapper?

Reply:
I just consistently call my components outside styled wrappers for `Wrapper` to avoid having to think and having the name I come up with have variant patterns.

Reply:
I can change that and make my rule component name + wrapper so it would be

`TabsButtonWrapper`

Reply:
As you also mentioned you would like styled components in a file of it's own this would be sorted as it could then easily become

`Styled.Button`